tests/repositories/fixtures/pypi.org/generate.py

```diff
@@ -518,6 +523,30 @@ def bdist_check(filename: str) -> bool:
 
         return bdist_check
 
+    def process_metadata_file(self, link: Link) -> None:
+        # we enforce the availability of the metadata file
+        link._metadata = True
+
+        logger.info("Processing metadata file for %s", link.filename)
+
+        assert link.metadata_url is not None
+        response: requests.Response = self.pypi.session.get(
+            link.metadata_url, raise_for_status=False
+        )
+
+        if response.status_code != 200:
+            logger.info("Skipping metadata for %s", link.filename)
+            return None
+
+        metadata, _ = parse_email(response.content)
+        content = response.content.decode(encoding="utf-8").replace(
+            metadata["description"], ""
+        )
+
+        FIXTURE_PATH_REPOSITORIES_PYPI.joinpath(
+            "metadata", f"{link.filename}.metadata"
+        ).write_text(content, encoding="utf-8")
```

Comment:
`write_bytes()`

probably everything is utf8 encoded all along so it works out fine, but from a hash-preserving point of view working with bytes throughout should be reliable

Reply:
The dropping of the description is not security related, never was. We really just don't need that information and it's frustrating when it shows up in searches. 

As for hashes, are metadata file hashes used anywhere? The tests passing seem to indicate they are not.

The binary file hashes are recomputed because we empty out the source. And the new hashes are used in fixtures.

Reply:
hashes are used in the branch that relies on metadata instead of the json api - if the hash doesn't match the value that the simpl page gives, then it is ignored

Reply:
Ah okay. Updated to ensure hashes are updated along with binary hashes.

Reply:
Will drop the version specific json files and their generation once #9170 is merged.